pkg/util/pki/certificatetemplate.go

```diff
@@ -261,6 +261,26 @@ func CertificateTemplateFromCSR(csr *x509.CertificateRequest, validatorMutators
 		}
 	}
 
+	// Finally, we fix up the certificate template to ensure that it is valid
+	{
+		// If the certificate has an empty Subject, we set any SAN extensions to be critical
```

Comment:
A comment explaining *why* would have been helpful, with a link to the RFC.

https://github.com/golang/go/blob/7020beeffda464a91f8275c3e89487bafad98428/src/crypto/x509/x509.go#L1180-L1182

Reply:
Agreed, should be added and I should've required it in review. Sorry!